test/built-ins/Atomics/wait/symbol-for-value-throws.js

```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2018 Amal Hussein. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-map-objects
+description: Test passing a Symbol for the value param throws. This test case also confirms that ToNumber is called.
+info: |
+  Atomics.wait( typedArray, index, value, timeout )
+
+  3.Let v be ? ToInt32(value).
+    ...
+      1.Let number be ? ToNumber(argument).
+        Symbol --> Throw a TypeError exception.
+features: [ TODO ]
+---*/
+
+var sab = new SharedArrayBuffer(1024);
+var int32Array = new Int32Array(sab);
+
+assert.throws(TypeError, () => Atomics.wait(int32Array, Symbol('foo'), 0, 0));
```

Comment:
this is similar to `test/built-ins/Atomics/wait/not-an-object-throws.js` but now for `index`.

You may group some invalid abrupt completions here:

```js
var p = { // poisoned object
  get valueOf() { throw "Should not evaluate this code"; } // throwing a string message here, not Test262Error
};
assert.throws(TypeError, function() {
  Atomics.wait(int32Array, Symbol, p, p);
}, 'Symbol');

var obj;
obj = { valueOf() { throw new Test262Error('verify poisoned'); }};
assert.throws(Test262Error, function() {
  Atomics.wait(int32Array, obj, p, p);
}, 'passing a poisoned object using valueOf');

obj = { get [Symbol.ToPrimitive]() { throw new Test262Error('verify poisoned'); }};
assert.throws(Test262Error, function() {
  Atomics.wait(int32Array, obj, p, p);
}, 'passing a poisoned object using @@ToPrimitive');
```

Add Symbol.toPrimitive to the features list